pkgs/tools/security/yubihsm-connector/default.nix

```
@@ -0,0 +1,29 @@
+{ lib, libusb, buildGo117Module, fetchFromGitHub, pkg-config }:
+
+buildGo117Module rec {
```

```suggestion
{ lib, libusb, buildGoModule, fetchFromGitHub, pkg-config }:

buildGoModule rec {
```

Default is 1.17 so I'd swap it with this

---

This package is explicitly old and never going to move past 117, so I may as well plan ahead and keep it that way, because it'll just break in future otherwise.

---

I'm not sure the reasons for the preference between pinning buildGoModules in the package default.nix vs pinning in all-packages.nix. Typically it seems to be pinned in all-packages.nix
At the least you'll want to add a comment explaining why it's pinned

It probably makes sense in this case to pin to 1.17 but go is pretty good at building older go with the newer releases.
What will be your plan for if it fails to build once 1.17 is gone? Patch the package or remove it?


As a note I made a PR that should fix the inconsistent vendoring and updating some packages so that might extend it's life a bit if it gets merged